tests/transforms/test_get_unitary_matrix.py

```diff
@@ -0,0 +1,296 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the get_unitary_matrix transform
+"""
+from functools import reduce
+import pytest
+import numpy as np
+from gate_data import I, X, Y, Z, H, S, CNOT
+import pennylane as qml
+
+from pennylane.transforms.get_unitary_matrix import get_unitary_matrix
+
+# test non-parametric single qubit gates
+nonparam_1qubit_op_list = [qml.PauliX, qml.PauliY, qml.PauliZ, qml.Hadamard, qml.S, qml.T, qml.SX]
+
+
+@pytest.mark.parametrize("op", nonparam_1qubit_op_list)
+@pytest.mark.parametrize("wire", [0, 1, 2])
+def test_get_unitary_matrix_nonparam_1qubit_ops(op, wire):
+    """Check the matrices for different nonparametrized single-qubit gates, which are acting on different qubits in a space of three qubits."""
+    wires = [0, 1, 2]
+
+    def testcircuit(wire):
+        op(wires=wire)
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix(wire)
+
+    if wire == 0:
+        expected_matrix = np.kron(op(wires=wire).matrix, np.eye(4))
+    if wire == 1:
+        expected_matrix = np.kron(np.eye(2), np.kron(op(wires=wire).matrix, np.eye(2)))
+    if wire == 2:
+        expected_matrix = np.kron(np.eye(4), op(wires=wire).matrix)
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+# Test a circuit containing multiple gates
+def test_get_unitary_matrix_multiple_ops():
+    """Check the total matrix for a circuit containing multiple gates. Also checks that non-integer wires work"""
+    wires = ["a", "b", "c"]
+
+    def testcircuit():
+        qml.PauliX(wires="a")
+        qml.S(wires="b")
+        qml.Hadamard(wires="c")
+        qml.CNOT(wires=["b", "c"])
+
+    get_matrix = get_unitary_matrix(testcircuit, wires)
+    matrix = get_matrix()
+
+    expected_matrix = np.kron(I, CNOT) @ np.kron(X, np.kron(S, H))
+
+    assert np.allclose(matrix, expected_matrix)
+
+
+def test_get_unitary_matrix_more_ops():
+    "Test some more operators. Same circuit as test_single_qubit_fusion_multiple_qubits() in test_single_qubit_fusion.py"
+    wires = ["a", "b"]
```

Comment:
Is this useful?

Reply:
Maybe not! I just wanted to test a lot of things, and saved it in the test file just in case. It can certainly be removed if you think it's superfluous.